container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -0,0 +1,345 @@
+# Copyright (c) 2020, Intel Corporation
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    * Redistributions of source code must retain the above copyright notice,
+#      this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#    * Neither the name of Intel Corporation nor the names of its contributors
+#      may be used to endorse or promote products derived from this software
+#      without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""Gateway IOT Device OnBoarding Framework."""
+
+import os.path
+import glob
+import argparse
+import subprocess
+import shutil
+import tempfile
+from collections import defaultdict
+from git import Repo
+import yaml
+NAME_SPACE = None
+
+
+def find_directory_name(dirpath):
+    """Return list of subdirectory paths for give directory."""
+    dirlist = []
+    for root, directories, _filename in os.walk(dirpath):
+        for dirs in directories:
+            dirlist.append(os.path.join(root, dirs))
+    return dirlist
+
+
+def check_deployment_status(appname):
+    """Return application deployment status in a given k8s node."""
+    cmd = "kubectl -n"+NAME_SPACE+" get deployment "+appname+" | wc -l"
+    process = subprocess.Popen(
+        cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    check_deploy_status = process.communicate()[0].decode('utf-8').strip()
+
+    # Cosmetic extraction of data from subprocess command
+    if '\n' in check_deploy_status:
+        check_deploy_status = check_deploy_status.splitlines()[1]
+    return check_deploy_status
```

Comment:
Returning `0`/`1` values is a little unpythonic. Maybe you can change it to a boolean, (or enum if more than 2 status are possible). If this feel overenginered, you should document possible return values in the docstring.

Reply:
agreed, done.